BranchMetrics/cpp-branch-deep-linking-attribution
# SDK-632 -- Disable Tracking option does not persist throughout multiple sessions.

## Reference
SDK-632 -- Disable Tracking option does not persist throughout multiple sessions.

## Description
We were not saving the Tracking State to shared preferences.   

## Testing Instructions
1. Disable tracking
2. Send an open request and verify it's unsuccessful
3. Close the app
4. Open it again
5. Send another Open Request

Expected Results:  Verify the request was *not* sent, nor should any other requests be sent until Enable Tracking is sent.

## Risk Assessment [`HIGH`]
Lots and lots of changes happened to make this work.


## Reviewer Checklist (To be checked off by the reviewer only)

- Correctness & Style
